mitodl/mitxonline
# Resolve issue when repairing user's edx synchronised records

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/1493

#### What's this PR do?
An issue occurs when we attempt to repair mitxonline User's edx OAuth accounts.  This issue occurs when a user meets the following criteria:
- User has an `OpenEdxUser` record in mitxonline.
- No user records exist for the user in Edx.

#### How should this be manually tested?
1. Have mitxonline and devstack up and running locally, and connected.
2. You will need to run a `makemigrations` in mitxonline.
3. Create a user through the mitxonline UI.
4. Delete the user's record, created during step 2, from devstack: http://localhost:18000/admin/auth/user/
5. From the Django shell, run:
```
from users.models import User
from openedx import api
user = User.objects.get(id=<USER_ID_FROM_STEP_2>)
result = api.repair_faulty_edx_user(user)
```

6. Verify that a user record is created in devstack: http://localhost:18000/admin/auth/user/
7. Verify that the OpenEdxUser record in mitxonline is marked as synced: http://mitxonline.odl.local:8013/admin/openedx/openedxuser/
